sklearn/ensemble/_forest.py

```diff
@@ -915,14 +971,27 @@ def _set_oob_score_and_attributes(self, X, y):
             The data matrix.
         y : ndarray of shape (n_samples, n_outputs)
             The target matrix.
+        sample_weight : ndarray of shape (n_samples,)
+            Sample weights.
         """
-        self.oob_prediction_ = super()._compute_oob_predictions(X, y).squeeze(
-            axis=1
+        self.oob_prediction_, oob_importances_ = (
```

Comment:
It would be enough to the final underscore when creating the attribute (this is the convention for fitted attribute).

```suggestion
        self.oob_prediction_, oob_importances = (
```